tools/nodejs/test/data_type_support.test.js

```diff
@@ -51,12 +53,39 @@ describe("data type support", function () {
       }
     );
   });
+
+  it("supports STRUCT values with NULL", function (done) {
+    db.run("CREATE TABLE struct_table (s STRUCT(a VARCHAR, b BOOLEAN))");
```

Comment:
Could we perhaps also add a test with recursive structures with NULLs at various levels, e.g. something like:

```sql
CREATE TABLE recursive_struct AS SELECT [
   { 'a': 42, 'b': [1, 2, 3]},
   NULL,
   { 'a': NULL, 'b': [4, NULL, 6]},
   {'a': 43, 'b': NULL}
] l UNION ALL SELECT NULL
```